guides/v2.1/test/integration/annotations/magentoDataFixture.md

```diff
@@ -0,0 +1,195 @@
+- Tests inherit test case fixtures, unless a test has its own specifically declared fixtures.
+- Declarations on the test case level don't affect tests with their own fixture annotation.
+
+The responsible algorithm uses the following logic:
+
+- If `@magentoDataFixture` appears in a doc comment for a particular test, the framework reverts all applied fixtures (if any) and applies the prescribed method-level ones. Upon a test completion, test-level fixtures are reverted.
+- If `@magentoDataFixture` appears in a doc comment for an entire test case, the framework applies the prescribed fixtures before every test of the test case, which declares no method-level fixtures, if class-level fixtures have not been applied yet (or have been already reverted). After the final test of the test case, all class-level fixtures are reverted.
+The integration testing framework uses DB transactions to revert applied fixtures.
+
+### Fixture Rollbacks
+
+Fixtures that perform solely database operations can be automatically reverted.
+If a fixture performs anything besides database operations (creates files, for instance) the custom rollback logic should be provided along with a fixture.
+
+Rollbacks execute after all database transactions have been reverted.
+
+Like fixtures, fixture rollbacks can be implemented as PHP files or methods.
+To function properly, a fixture rollback must be the same type (script or method) as the corresponding fixture, and the name should be based on the fixture name.
+
+Rollback script files are named with the suffix _rollback and stored in the same folder as the corresponding fixture.
+Rollback methods are of the same class as the corresponding fixture, and are named with the suffix Rollback.
+For example:
+
+Fixture/Rollback type|Fixture name|Rollback name
+---|---|---
+Script|`Magento/Catalog/_files/categories.php`|`Magento/Catalog/_files/categories_rollback.php`
+Method|`Magento_Catalog_Model_ProductTest::prepareProduct`|`Magento_Catalog_Model_ProductTest::prepareProductRollback`
+
+### Restrictions
+
+Relying on and modifying application state from within a fixture is prohibited.
+
+The limitation is dictated by the compatibility requirements with the [application isolation annotation][magentoAppIsolation], which may reset application state at any time.
+
+For example, the following fixture results passing to curresponding test is prohibited:
+
+> Prohibited data passing through the global registry
+
+```php?start_inline=1
+$user = Mage::getModel('Magento_User_Model_User');
+$user->setFirstname('Dummy')
+    ->setLastname('Dummy')
+    ->setEmail('dummy@dummy.com')
+    ->setUsername('dummy_username')
+    ->setPassword('dummy_password1')
+    ->save();
+Mage::register('_fixture/user_id', $user->getId()); // incompatible with eventual app isolation
+```
+
+#### Session fixtures
+
+Session fixtures must not use `Mage::getSingleton()` to instantiate a necessary session model.
```

Comment:
This session refers to Magento 1. The main idea is still valid for Magento 2.

Reply:
changed `Mage::getSingleton()` to `::getSingleton`
Is it correct?